dateutil/parser.py

```diff
@@ -725,11 +726,20 @@ def _parse(self, timestr, dayfirst=None, yearfirst=None, fuzzy=False,
                     # Token is a number
                     len_li = len(l[i])
 
-                    if (len(ymd) == 3 and len_li in (2, 4) and
-                        res.hour is None and
-                        (i + 1 >= len_l or
-                         (l[i + 1] != ':' and
-                          info.hms(l[i + 1]) is None))):
+                    hms_idx = _find_hms_idx(i, l, info, allow_jump=True)
```

Comment:
I think that was an option used in an earlier implementation and can now be removed.

In the loop-based implementation being replaced, only the first step of this check is allowed to look past a space for a h/m/s label.  This implementation should make that irrelevant.